patches/api/0417-add-PlayerCreateMapEvent.patch

```diff
@@ -0,0 +1,65 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: chmodsayshello <chmodsayshello@hotmail.com>
+Date: Tue, 2 May 2023 18:11:50 +0200
+Subject: [PATCH] add PlayerCreateMapEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerCreateMapEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerCreateMapEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ea6a9b7d98161bfaf41c397d30b23af1ba2dffa0
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerCreateMapEvent.java
+@@ -0,0 +1,53 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a player generates a new map.
++ * This happens when right-clicking on a non filled map
++ * <p>
++ * Cancelling it will result in the map not getting generated.
++ * That way the item won't convert into a filled map...
++ */
++public class PlayerCreateMapEvent extends PlayerEvent implements Cancellable {
++    private boolean cancelled = false;
++    private final ItemStack mapStack;
++    private static final HandlerList handlers = new HandlerList();
```

Comment:
Should use the java standard for static final fields, I.e. `HANDLER_LIST` probably

Reply:
At first, I did that, however, I noticed that almost every class within that package uses calls it `handlers` instead

Reply:
https://github.com/PaperMC/Paper/pull/9176#discussion_r1186716577 - that's just bukkit's weird practice